jvm-packages/xgboost4j-spark/src/main/scala/ml/dmlc/xgboost4j/scala/spark/params/GeneralParams.scala

```diff
@@ -147,3 +169,47 @@ trait HasBaseMarginCol extends Params {
   /** @group getParam */
   final def getBaseMarginCol: String = $(baseMarginCol)
 }
+
+trait HasNumClass extends Params {
+
+  /**
+   * number of classes
+   */
+  final val numClass = new IntParam(this, "numClass", "number of classes")
+
+  /** @group getParam */
+  final def getNumClass: Int = $(numClass)
+}
+
+private[spark] trait ParamMapFuncs extends Params {
+
+  def XGBoostToMLlibParams(xgboostParams: Map[String, Any]): Unit = {
```

Comment:
for these two functions, I think they desire the additional logic which checks if the user passed in any params not existing (which usually means the user has some typo there...), so that we can stop training early

Reply:
Good point, we have several params legal check related fix in following PRs.